src/wire/icmpv6.rs

```diff
@@ -68,6 +69,7 @@ impl Message {
     ///
     /// [MLD]: https://tools.ietf.org/html/rfc3810
     pub fn is_mld(&self) -> bool {
+        #[allow(clippy::match_like_matches_macro)]
```

Comment:
Given that our MRSV is 1.36 and `matches!` is not supported there, it'd make more sense to allow it for the entire crate (`#![allow(clippy::match_like_matches_macro)]` in `lib.rs`)